File: P_4/Napakalaki_Ruby_P4/lib/death_bad_consequence.rb

module NapakalakiRuby
    
require_relative 'bad_consequence'
require_relative 'numeric_bad_consequence'
require_relative 'player'
    
class DeathBadConsequence < NumericBadConsequence
    
    public_class_method :new
    
    # Constructor
    def initialize (text)
        super(text, Player.get_max_level, BadConsequence.getMaxTreasures,
            BadConsequence.getMaxTreasures)
    end
end

end

File: P_4/Napakalaki_Java_P4/src/napakalaki/DeathBadConsequence.java

package napakalaki;

/**
 *
 * @author ixjosemi
 */
public class DeathBadConsequence extends NumericBadConsequence{
    
    // Constructor
    public DeathBadConsequence(String text){
        super(text,Player.MAXLEVEL, BadConsequence.MAXTREASURES, 
                BadConsequence.MAXTREASURES);

    }
    
}